test/prism_regression/literal_integer.rb

```diff
@@ -7,3 +7,27 @@
 
 # ~ should not be a method call
 ~42
+
+# Hexadecimal literal
+0xcafe
+0XCAFE
++0xcafe
+-0xcafe
+
+# Decimal literal
+0d123
+0D123
++0d123
+-0d123
+
+# Octal literal
+0o777
+0O777
++0o777
+-0o777
```

Comment:
Octals can use `o`, `O`, but confusingly also `O`. E.g.

```ruby
irb(main):001> 00777
511
```

```suggestion
0o777
0O777
+0o777
-0o777
+00777
-00777
```